### PR TITLE
Manage Qserv version at configuration

### PR DESCRIPTION
**Pull request by fjammes**
Rely on eupspkg to detect correct Qserv version

### CODE REVIEW

File: admin/bin/qserv-configure.py [hunk 1]
```diff
@@ -321,8 +324,8 @@ def run(self):
                 # used in templates targets comments
                 config['qserv']['meta_config_file'] = self._meta_config_file
 
-            except ConfigParser.NoOptionError as exc:
-                _LOG.fatal("Missing option in meta-configuration file: %s", exc)
+            except ConfigParser.NoOptionError as exception:
+                _LOG.fatal("Missing option in meta-configuration file: %s", exception)
```
Comment (vaikuntht) [inline admin/bin/qserv-configure.py]: "exception" is still "exc" in other places of the code. We should either change them all to "exception" or leave it as "exc."

File: admin/bin/qserv-version.sh [hunk 2]
```diff
@@ -20,10 +20,53 @@
...
-
-# Return Qserv version number
-# Used to determine if Qserv run dir is compliant with Qserv version
+# Return Qserv version or release number
 
 # @author  Fabrice Jammes, IN2P3
-QSERV_VERSION="2016_07"
-echo "${QSERV_VERSION}"
+
+QSERV_RELEASE="2016_07"
```
Comment (vaikuntht) [inline admin/bin/qserv-version.sh]: Will this have to be changed every month?

File: admin/bin/qserv-version.sh [hunk 3]
```diff
@@ -20,10 +20,53 @@
...
+  cat << EOD
+
+Usage: `basename $0` [options]
+
+  Available options:
+    -h          this message
+    -R          print release number
+
+  Print Qserv (eups-based) version number,
+  or Qserv release on which this version is based
```
Comment (vaikuntht) [inline admin/bin/qserv-version.sh]: What is the exact difference between the Qserv "version number" and "release?"

File: admin/python/lsst/qserv/admin/configure.py [hunk 4]
```diff
@@ -369,8 +227,149 @@ def has_configuration_step(steps):
     """
     return bool(intersect(steps, CONFIGURATION_STEPS))
 
+class Templater(object):
+
+    def __init__(self, qserv_version):
+
```
Comment (vaikuntht) [inline admin/python/lsst/qserv/admin/configure.py]: Remove empty line

File: admin/python/lsst/qserv/admin/configure.py [hunk 5]
```diff
@@ -369,8 +227,149 @@ def has_configuration_step(steps):
...
+            # all other files are configuration files
+            os.chmod(file, 0o660)
+
+    def _initTemplateParams(self):
+        """ Compute templates parameters from Qserv meta-configuration file
+            from PATH or from environment variables for products not needed during build
+        """
+
+        if self._templateParams is None:
+
```
Comment (vaikuntht) [inline admin/python/lsst/qserv/admin/configure.py]: Remove empty line

File: admin/python/lsst/qserv/admin/configure.py [hunk 6]
```diff
@@ -369,8 +227,149 @@ def has_configuration_step(steps):
...
+                    basename in script_list):
+            os.chmod(file, 0o760)
+        elif basename in SECRET_FILES:
+            os.chmod(file, 0o600)
+        else:
+            # all other files are configuration files
+            os.chmod(file, 0o660)
+
+    def _initTemplateParams(self):
+        """ Compute templates parameters from Qserv meta-configuration file
```
Comment (vaikuntht) [inline admin/python/lsst/qserv/admin/configure.py]: Comment block syntax should be:
```
def func():
   """
   Comment
   """
```

File: admin/python/lsst/qserv/admin/configure.py [hunk 7]
```diff
@@ -369,8 +227,149 @@ def has_configuration_step(steps):
...
+            else:
+                comment_mono_node = ''
+
+            scisql_dir = os.environ.get('SCISQL_DIR')
+            if scisql_dir is None:
+                _LOG.fatal(
+                    "sciSQL install : sciSQL is missing, please install it and set SCISQL_DIR environment variable.")
+                sys.exit(1)
+
+            # find python executable in $PATH
```
Comment (timj) [inline admin/python/lsst/qserv/admin/configure.py]: Does the python have to be in the path or can it be the python that is running this code?
```python
python_bin = sys.executable
```
would be simpler.

File: admin/bin/qserv-version.sh (hunk 2 above)
Comment (fjammes) [inline admin/bin/qserv-version.sh]: Yes, Fritz should do that when publishing release.

File: admin/bin/qserv-version.sh (hunk 3 above)
Comment (fjammes) [inline admin/bin/qserv-version.sh]: 'version number' is computed by eups and depends on git tag, 'release' is set by hand each month.

File: admin/python/lsst/qserv/admin/configure.py (hunk 7 above)
Comment (fjammes) [inline admin/python/lsst/qserv/admin/configure.py]: ok

**Pull request merged**